src/it/scala/nl/wbaa/gargoyle/proxy/S3ProxyItTest.scala

```diff
@@ -0,0 +1,188 @@
+package nl.wbaa.gargoyle.proxy
+
+import java.io.{File, RandomAccessFile}
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.Uri.{Authority, Host}
+import com.amazonaws.services.s3.AmazonS3
+import com.whisk.docker.impl.spotify.DockerKitSpotify
+import com.whisk.docker.scalatest.DockerTestKit
+import nl.wbaa.gargoyle.proxy.config.{GargoyleHttpSettings, GargoyleStorageS3Settings}
+import nl.wbaa.testkit.docker.DockerCephS3Service
+import nl.wbaa.testkit.s3sdk.S3SdkHelpers
+import org.scalatest._
+
+import scala.collection.JavaConverters._
+import scala.concurrent.Future
+
+class S3ProxyItTest extends AsyncWordSpec with DiagrammedAssertions
+  with DockerTestKit
+  with DockerKitSpotify
+  with DockerCephS3Service
+  with S3SdkHelpers {
+  private[this] final implicit val system: ActorSystem = ActorSystem.create("test-system")
+
+  // Settings for tests:
+  //  - Force a random port to listen on.
+  //  - Explicitly bind to loopback, irrespective of any default value.
+  private[this] val gargoyleHttpSettings = new GargoyleHttpSettings(system.settings.config) {
+    override val httpPort: Int = 0
+    override val httpBind: String = "127.0.0.1"
+  }
+
+  // Settings to connect to S3 storage, we have to wait for the docker container to retrieve the exposed port
+  private[this] lazy val gargoyleStorageS3SettingsFuture: Future[GargoyleStorageS3Settings] =
+    cephContainer.getPorts()(docker = dockerExecutor, ec = dockerExecutionContext)
+      .map { portMapping =>
+        println(portMapping)
```

Comment:
logger is better

Reply:
removed as it was not intended to keep here